### PR TITLE
DrivAerML: 600 batches/epoch + weight decay 1e-2 compound

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Compound the 600-batch improvement (PR #2645, 11.97%) with stronger weight decay (1e-2 vs the AdamW default of ~1e-4). With more data per epoch, stronger regularization may improve generalization by preventing the model from memorizing the most frequently-seen car configurations. WD=1e-2 showed promising convergence on TandemFoil with Lion optimizer — here we test whether the same regularization boost transfers to DrivAerML with AdamW and 600 batches/epoch.

## Instructions

Add `--weight-decay 1e-2` to the PR #2645 winning configuration. All other settings are identical to the current best.

**Full command to run:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 \
  --weight-decay 1e-2 --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 600 --max-eval-batches 200 \
  --agent norman --wandb-name "norman/drivaerml-600batch-wd1e2" \
  --wandb-group "drivaerml-600batch-compounds"
```

## Baseline to beat

- **Primary metric:** `val_primary/surface_rel_l2_pct` (lower is better)
- **Current best val:** 11.97% (PR #2645, 600 batches/epoch, default WD)
- **Current best test:** 13.03% (PR #2645)
- **W&B run:** dar47nwl
- **Key config:** 4L/256d, AdamW lr=5e-4, cosine T_max=30, Fourier, no-EMA, batch-size=1, 50k surface points, 600 train batches/epoch

## W&B group

`drivaerml-600batch-compounds` — groups experiments compounding other improvements on top of the 600-batch baseline.

## Expected outcome

If WD=1e-2 is too strong, training may slow or diverge. If well-tuned, we expect val below 11.97%. Monitor val/loss and training loss carefully — if training loss diverges, note the epoch. Report best val metric, the epoch at which it occurred, and whether training appeared stable.

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2672 because ADVISOR: Closing — DrivAerML baseline shifted to 5.73% (4L/384d, PR #2602). The 4L/256d+600batch config cannot compete. Redirecting to 4L/384d experiments..
